### PR TITLE
Add D implementation of monte carlo

### PR DESCRIPTION
**Pull request by scarface-one**
Its a fairly straight forward implementation. Simply run with `rdmd monte_carlo.d`

### CODE REVIEW

File: chapters/monte_carlo/code/d/monte_carlo.d [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+///Calculate pi using monte carlo
+real monteCarloPI(ulong n)
+{
+    import std.algorithm : count;
+    import std.random : Random, uniform01, unpredictableSeed;
+    import std.range : generate, take;
+    import std.typecons : tuple;
+
+    auto rnd = Random(unpredictableSeed);
+    return generate(() => tuple!("x", "y")(rnd.uniform01, rnd.uniform01))
```
Comment (Gustorn) [inline chapters/monte_carlo/code/d/monte_carlo.d]: I think we should make this part a little more explicit, something like:

```d
    auto radius = 1.0;
    auto piCount = generate(() => tuple!("x", "y")(rnd.uniform01, rnd.uniform01))
        .take(n)
        .count!(a => inCircle(a.x, a.y, radius));
    return 4.0 * piCount / (n * radius * radius);
```

I definitely think radius should be a variable just or clarity's sake and with that change, jamming the return value into one expression seems a bit over the top.

Comment (scarface-one) [reply chapters/monte_carlo/code/d/monte_carlo.d]: I can do the return bit, but its worth noting that the Julia code is being refactored to remove the radius variable entirely. I think that would be better.

Comment (Gustorn) [reply chapters/monte_carlo/code/d/monte_carlo.d]: Removing `radius` also works but I'd still introduce the separate variable for `piCount`.

Comment (scarface-one) [reply chapters/monte_carlo/code/d/monte_carlo.d]: done

**Pull request merged**